x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CreateSnapshotStep.java

```diff
@@ -61,7 +62,17 @@ public void onResponse(Boolean complete) {
 
             @Override
             public void onFailure(Exception e) {
-                listener.onFailure(e);
+                if (snapshotAlreadyCreated(e)) {
+                    logger.warn(e.getMessage());
+                    // we treat a snapshot that was already created before this step as an incomplete snapshot. This means that the
+                    // execution will be successful from ILM's perspective and it will go to the next step, which is cleaning the incomplete
+                    // snapshot and try again. This scenario is triggered by a master restart or a failover which can result in a double
+                    // invocation of this step.
+                    onResponseResult.set(false);
```

Comment:
What about something like `onResponse(false)` so that you reroute to the other handler rather than repeating some of the same logic? The comment here could be simpler then, too.

Reply:
Very good suggestion, I will adjust it.